builtin/providers/aws/resource_aws_instance.go

```diff
@@ -469,6 +469,13 @@ func resourceAwsInstanceRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("public_ip", instance.PublicIpAddress)
 	d.Set("private_dns", instance.PrivateDnsName)
 	d.Set("private_ip", instance.PrivateIpAddress)
+
+	if instance.IamInstanceProfile != nil {
+		d.Set("iam_instance_profile", iamInstanceProfileArnToName(instance.IamInstanceProfile.Arn))
+	} else if _, ok := d.GetOk("iam_instance_profile"); ok {
```

Comment:
Hmm, interesting... I'd never noticed that Terraform made any distinction there before. I thought it always just wrote everything out in all cases, but I guess you learn something new every day! Thanks!